confdb: avoid syslog message when no domains are enabled

pbrezina opened the pull request:
This syslog message would also appear when calling other tools like 
sss_cache which is confusing. We return specific error code instead and let
the error be syslogged in the monitor in monitor.c:main (this is already
implemented).

Resolves: https://github.com/SSSD/sssd/issues/6387

---

monitor: read all enabled domains in add_implicit_services

Reading sssd/domains option is no longer sufficient since domains can be
enabled through domain/enabled.

sumit-bose commented:
Hi,

the message in the system logs is gone, but there is still `[sss_cache] [confdb_get_domains] (0x0010): No domains configured, fatal error!` on stderr. I guess this should be suppressed as well to not confuse other programs calling `sss_cache`? Have you checked with the original reporters if the patch already fixes their issue.

bye,
Sumit

pbrezina commented:
Hmm, that's because I also changed the severity from SSSDBG_MINOR_FAILURE to SSSDBG_FATAL_FAILURE, since it kind of make sense because it is a fatal error. But it made it visible in the tools so I added another patch that drops the level to critical and prints the message in the monitor as fatal error as well.

The reported confirmed that the previous batch of patches fixed their issue (they only care about syslog, sss_cache is run by shadow-utils and not directly).

alexey-tikhonov commented:
> But it made it visible in the tools so I added another patch that drops the level to critical and prints the message in the monitor as fatal error as well.

FWIW, default log level is `SSSDBG_OP_FAILURE`, so `SSSDBG_CRIT_FAILURE` will be printed by default as well.

aplopez commented on src/util/util_errors.h:
```diff
@@ -85,6 +85,7 @@ enum sssd_errors {
     ERR_INPUT_PARSE,
     ERR_NOT_FOUND,
     ERR_DOMAIN_NOT_FOUND,
+    ERR_NO_DOMAIN_ENABLED,
```
I have just one minor concern here.
So far `ERR_INVALID_FILTER` was error _X_. Now error _X_ is `ERR_NO_DOMAIN_ENABLED` and `ERR_INVALID_FILTER` (and all the following errors) where increased by one.
Is it possible that these values arrive somehow to the user (for instance, `main()` returns these error codes)? In that case, this modification changes that tool's behavior because previously it would return _Y_ is now it will return _Y+1_ (if _Y_ >= _X_).
Is this a problem?

pbrezina replied:
Not a problem. We rather place it to a relevant position. sss_cache does not have any documented exit codes.

sumit-bose reviewed (approved):
Hi,

thanks for the update, ACK.

bye,
Sumit

aplopez reviewed (approved):
ACK, Thanks.

pbrezina commented:
I just added a release note to the first commit message:
```
    confdb: avoid syslog message when no domains are enabled

    This syslog message would also appear when calling other tools like
    sss_cache which is confusing. We return specific error code instead
    and let the error be syslogged in the monitor in monitor.c:main (this
    is already implemented).

    Resolves: https://github.com/SSSD/sssd/issues/6387

    :fixes: A regression when running sss_cache when no SSSD domain is
      enabled would produce a syslog critical message was fixed.

```

sumit-bose commented:
> I just added a release note to the first commit message:
> 

Hi,

thanks, ACK.

bye,
Sumit

pbrezina commented:
Pushed PR: https://github.com/SSSD/sssd/pull/6388

* `master`
    * df55b1f16ea2fb5e56f0fe69419904d50b7e2476 - confdb: chande debug level when no domain are found in confdb_get_domains
    * 64c22dd1c4f79f953b879fc167b535de928f4bfd - sss_cache: use ERR_NO_DOMAIN_ENABLED instead of ENOENT
    * 4da861368b88c03b22993f95de5b508ad5637c25 - monitor: read all enabled domains in add_implicit_services
    * b38fdc8185fcd6a2e5d4b483d3119964f9922070 - confdb: avoid syslog message when no domains are enabled
* `sssd-2-8`
    * ab7cbb5f0f74e5061136cf632f487d7a8c023272 - confdb: chande debug level when no domain are found in confdb_get_domains
    * a6312c46daebcf39b9331cc2c99f5abfe25f64d0 - sss_cache: use ERR_NO_DOMAIN_ENABLED instead of ENOENT
    * 1df029470df8af58cf63e41bdad7b8882b141ac3 - monitor: read all enabled domains in add_implicit_services
    * a6d521458460593afc7585f2d269aa4c6f326ecb - confdb: avoid syslog message when no domains are enabled
